notebooks/search-analysis/search_analysis.ipynb

```diff
@@ -0,0 +1,399 @@
+   "source": [
+    "relevancy_judgements[\"best_bet\"] = relevancy_judgements['search_term'].map(\n",
+    "            lambda x: bool(regex.search(best_bets_regex, x, re.IGNORECASE)) if isinstance(x, str) else False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "search_df['date'] = search_df['search_timestamp'].map(lambda x:x[:7])\n",
+    "search_df['tokens'] = search_df['search_term'].map(lambda x:x.split())\n",
+    "search_df['query_length'] = search_df['tokens'].map(len)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "avg_len=search_df[['date','query_length']].groupby('date').mean().reset_index()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Query length over time\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "avg_len=search_df[['date','query_length']].groupby('date').mean().reset_index()\n",
+    "%matplotlib inline\n",
+    "plt.style.use('ggplot')\n",
+    "plt.figure(figsize=(10,5))\n",
+    "plt.plot(avg_len['date'], avg_len['query_length'], color='green')\n",
+    "plt.xticks(rotation=45, ha=\"right\")\n",
+    "plt.ylabel(\"Average length\")\n",
+    "\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Query length hist (looks exponentially)\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline\n",
+    "fig, ax = plt.subplots(1, 1)\n",
+    "plt.style.use('ggplot')\n",
+    "plt.hist(search_df['query_length'], color='green',bins=300,density=True)\n",
+    "plt.xlabel(\"Query length\")\n",
+    "plt.xticks(rotation=45, ha=\"right\")\n",
+    "plt.ylabel(\"Density\")\n",
+    "plt.xlim(1,15)\n",
+    "plt.xticks(np.arange(1, 15, step=1))\n",
+    "ax.axvline(x=np.quantile(search_df['query_length'],0.75))\n",
+    "fig.set_size_inches(10,5)\n",
+    "\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "# Specific service: query names (or almost names) specific service or form (regardless of whether that service is on gov.uk)\n",
+    "# Specific guidance: query names specific guidance/stats doc on gov uk\n",
+    "# Guidance: query references topic user presumably wants guidance about\n",
+    "# Contact: query explicitly asks for contact, email, phone number, chat etc\n",
+    "# Unknown: mysterious!\n",
+    "\n",
+    "\n",
+    "# annotation = annotate(df['search_term'],options=['specific_service','specific_guidance','guidance','contact','unknown'])\n",
+    "# query_cats =pd.DataFrame(annotation, columns=['query','category'])\n",
+    "query_cats = pd.read_csv(PROCESSED_DATA + '/query_categories.csv')\n",
```

Comment:
Could we call this file from within the `data/` folder in the repo instead for better reproducibility?

Reply:
Also, good to have some notes on how this file was created. I presume it was govNER?